packages/fiber/src/three-types.ts

```diff
@@ -177,12 +170,27 @@ export type SpriteMaterialProps = MaterialNode<THREE.SpriteMaterial, [THREE.Spri
 export type RawShaderMaterialProps = MaterialNode<THREE.RawShaderMaterial, [THREE.ShaderMaterialParameters]>
 export type ShaderMaterialProps = MaterialNode<THREE.ShaderMaterial, [THREE.ShaderMaterialParameters]>
 export type PointsMaterialProps = MaterialNode<THREE.PointsMaterial, [THREE.PointsMaterialParameters]>
-export type MeshPhysicalMaterialProps = MaterialNode<THREE.MeshPhysicalMaterial, [THREE.MeshPhysicalMaterialParameters]>
-export type MeshStandardMaterialProps = MaterialNode<THREE.MeshStandardMaterial, [THREE.MeshStandardMaterialParameters]>
-export type MeshPhongMaterialProps = MaterialNode<THREE.MeshPhongMaterial, [THREE.MeshPhongMaterialParameters]>
-export type MeshToonMaterialProps = MaterialNode<THREE.MeshToonMaterial, [THREE.MeshToonMaterialParameters]>
+export type MeshPhysicalMaterialProps = Overwrite<
+  MaterialNode<THREE.MeshPhysicalMaterial, [THREE.MeshPhysicalMaterialParameters]>,
+  { sheenColor?: Color; attenuationColor?: Color; specularColor?: Color }
+>
+export type MeshStandardMaterialProps = Overwrite<
```

Comment:
I'm a bit confused, why isn't `emissive` coming from the `types` library? Is that because it's missing there, or....?

Reply:
These are all present in the types, but we redefine `Color` here. Maybe we can do this automatically on the base node type?

Reply:
can TS do this? pick props of a certain type and re-define them? that would be wonderful

Reply:
I guess material node could be used to form a utility type where if certain keys are apparent e.g. `emissive` you could then change that type to be `Color` and if it's not I believe `never` is the correct replacement.